core/src/main/scala/chisel3/Module.scala

```diff
@@ -382,6 +384,17 @@ package experimental {
       names
     }
 
+    /** Invokes _onModuleClose on HasIds found via reflection but not bound to hardware
```

Comment:
what is an example of a HasIds that would be found via reflection but not bound to hardware?

Reply:
It doesn't show up in the diff because all I did was make a symlink so that the same file is compiled with and without the plugin, but from [InstanceNameSpec](https://github.com/chipsalliance/chisel3/blob/2ed343e2305b7c22000f3f46fa81d73a369907eb/src/test/scala/chiselTests/InstanceNameSpec.scala#L9):
```scala
class InstanceNameModule extends Module {
  // ...
  val z = new Bundle {
    val foo = UInt(8.W)
  }
  // ...
}
```
It's unused, but after elaborating that module you can ask for `z.foo.pathName`.

Reply:
ah yeah, we talked about that IRZoomL. Ok, so this is handling the case that we said was weird, but people are apparently doing it, and we don't have a great way of deprecating it